### PR TITLE
Fancy Cigar in Aligner

### PR DESCRIPTION
**Pull request by DanielSnellings**
Added conversion to fancy cigar in aligner. Not sure if this will get merged or not, but I figured it would be good to put it up for discussion. Performance impact seems to be <5% depending on read with the majority of reads <2%. Uncomment the lines in toGiraf.go to test

### CODE REVIEW

File: simpleGraph/toGiraf.go [hunk 1]
```diff
@@ -81,9 +83,91 @@ func GraphSmithWatermanToGiraf(gg *SimpleGraph, read *fastq.FastqBig, seedHash m
...
 }
 
+func GirafToExplicitCigar(giraf *giraf.Giraf, graph *SimpleGraph) []*cigar.Cigar {
+	var answer []*cigar.Cigar
+	var seqIdx, refIdx, pathIdx int
+	refIdx = giraf.Path.TStart
+	var k, runLenCount int64
+
+	if giraf.Aln[0].Op == '*' {
+		return nil
```
Comment (craiglowe) [inline simpleGraph/toGiraf.go:103]: Do we want this to be nil?  '*' means unaligned read, right?  Do we want to keep the cigar as a '*' for unaligned or should we do nil?  This should probably be consistent across all cigars, right?

File: simpleGraph/toGiraf.go [hunk 2]
```diff
@@ -81,9 +83,91 @@ func GraphSmithWatermanToGiraf(gg *SimpleGraph, read *fastq.FastqBig, seedHash m
...
+	}
+
+	for i := 0; i < len(giraf.Aln); i++ {
```
Comment (craiglowe) [inline simpleGraph/toGiraf.go:106]: This looks really similar to cigar.MakeExplicit.  Can we call that?

File: simpleGraph/toGiraf.go (hunk 1 above)
Comment (edotau) [reply simpleGraph/toGiraf.go]: yes, this is not consistent throughout the code base but i think the slice of cigars should be set to nil if it is unaligned. When we print (to string and/or to file) we check if == nil, print *

File: cigar/fancy.go [hunk 3]
```diff
@@ -14,6 +14,11 @@ func MakeExplicit(cigar []*Cigar, sequence []dna.Base, reference []dna.Base) []*
 	var answer []*Cigar
 	var seqIdx, refIdx int
 	var k, runLenCount int64
+
+	if cigar[0].Op == '*' {
+		return []*Cigar{&Cigar{Op: '*'}}
```
Comment (craiglowe) [inline cigar/fancy.go:19]: I think Eric weighed in here saying that the original version (return nil) is correct and I think my earlier comment threw us off.  Seems like there are two ways to do it currently in the code, but we are going to try to converge on the "nil = unaligned"

Comment (DanielSnellings) [reply cigar/fancy.go]: Switched it back to nil

**Pull request merged**